src/Symfony/Component/Messenger/Transport/TransportFactoryInterface.php

```diff
@@ -17,9 +17,14 @@
  * Creates a Messenger transport.
  *
  * @author Samuel Roze <samuel.roze@gmail.com>
+ *
+ * @template T
```

Comment:
Right now, `T` would be of type mixed, which is _a little_ too wide. Moreover, we should use a more expressive template variable name and allow covariance.

```suggestion
 * @template-covariant TTransport of TransportInterface
```